test/functional/feature_datacarrier.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020-2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test datacarrier functionality"""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.script import (
+    CScript,
+    OP_RETURN,
+)
+from test_framework.wallet import (
+    MiniWallet
+)
+from test_framework.messages import (
+    CTxOut,
+)
+from test_framework.util import (
+    assert_raises_rpc_error,
+)
+
+class DataCarrierTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def test_transaction(self, tx, success):
+        tx_hex = tx.serialize().hex()
+
+        if success:
+            self.wallet.sendrawtransaction(from_node=self.nodes[0], tx_hex=tx_hex)
+            assert(tx.rehash() in self.nodes[0].getrawmempool(True))
+        else:
+            assert_raises_rpc_error(-26, "scriptpubkey", self.wallet.sendrawtransaction, from_node=self.nodes[0], tx_hex=tx_hex)
+
+    def create_null_data_transaction(self, script):
+        tx = self.wallet.create_self_transfer(fee_rate=0)["tx"]
+        tx.vout.append(CTxOut(nValue=0, scriptPubKey=script))
+        tx.vout[0].nValue -= tx.get_vsize()  # simply pay 1sat/vbyte fee
+        return tx
+
+    def run_test(self):
+        self.node = self.nodes[0]
+        self.wallet = MiniWallet(self.nodes[0])
```

Comment:
nit (feel free to ignore)
```suggestion
        self.wallet = MiniWallet(self.nodes[0])
        self.wallet = MiniWallet(self.node)
```

Reply:
The test now is using 3 nodes to avoid restarting, as suggested in https://github.com/bitcoin/bitcoin/pull/25792#pullrequestreview-1065378600. So `self.node ` is no longer being used. Thanks for the review.